teerex/src/lib.rs

```diff
@@ -446,6 +446,13 @@ impl<T: Config> Pallet<T> {
 	}
 
 	fn unregister_silent_workers(now: T::Moment) {
+		let minimum = now
+			.checked_sub(&T::MaxSilenceTime::get())
+			.unwrap_or_default()
+			.saturated_into::<u64>();
+		if minimum == 0 {
+			log::error!("Invalid time in unregister_silent_workers. Is the timestamp pallet properly configured?")
```

Comment:
The question is what to do here? I think a log message is at least better than ignoring the underflow.

Reply:
Timestamp has using system time under the hood. However, I am fine which catching this, but maybe we want to early return in this case?

Reply:
Indeed, silently ignoring is way worse.

I am unsure about the impact though. Does it make sense to continue execution in case the timestamp pallet is not configured properly? If not, then I suggest bailing out here.

Reply:
Yeah, i mean this is very unlikely scenario. The timestamp is set by the timestamp inherent. Inherents are verified by all the other nodes when importing a block to be plausible because, another node can only verify it the timestamp inherent of a block is in reasonable range, but not verify it exactly. Hence, having a wrong value here, would mean that a substantial fraction of the blockchain nodes of the network have a wrong system time configured.

Reply:
Yeah, the most likely scenario is a unit test (how I found it) or some other small test setup I would say. Should not happen in production.